src/main/groovy/pl/allegro/tech/build/axion/release/domain/VersionResolver.groovy

```diff
@@ -44,27 +46,72 @@ class VersionResolver {
         GitRepository repository = (GitRepository) this.repository
         Pattern releaseTagPattern = ~/^${tagProperties.prefix}.*/
         Pattern nextVersionTagPattern = ~/.*${nextVersionProperties.suffix}$/
-
-        TagsOnCommit tags = repository.latestTags(releaseTagPattern)
-        TagsOnCommit previousTags = tags;
-
-        Map currentVersionInfo = versionFromTags(tags.tags, false, nextVersionTagPattern, versionFactory)
+				
+        Map currentVersionInfo = null
+				Map previousVersionInfo = null
+				boolean isHead = false
+				
+				if (versionFactory.versionProperties.useHighestVersion) {
+					LinkedHashMap<String, List<String>> allTaggedCommits = repository.allTaggedCommits(releaseTagPattern, null, true)
```

Comment:
i would put both branches in separate methods, names would provide some more documentation